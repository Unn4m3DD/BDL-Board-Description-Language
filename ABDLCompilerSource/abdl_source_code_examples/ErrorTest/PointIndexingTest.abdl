main:
    let i = 0;
    let s = "";
    let p = [0, 0];

    let v0 = i[i];
    let v1 = i[p];
    let v2 = i[s];

    let v3 = p[i];
    let v4 = p[p];
    let v5 = p[s];

    let v6 = s[i];
    let v7 = s[p];
    let v8 = s[s];
end main;