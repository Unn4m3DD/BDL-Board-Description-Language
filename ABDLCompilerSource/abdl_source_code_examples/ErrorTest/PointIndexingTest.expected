Only points are indexable (6:13): i is of type int
Only points are indexable (7:13): i is of type int
Indexes must be integers (0 or 1)(7:13): p is of type point
Only points are indexable (8:13): i is of type int
Indexes must be integers (0 or 1)(8:13): s is of type string